core/src/rpc.rs

```diff
@@ -425,17 +433,42 @@ impl JsonRpcRequestProcessor {
             .map(|(_, status)| status)
     }
 
-    pub fn get_signature_statuses_with_commitment(
+    pub fn get_signature_statuses(
         &self,
         signatures: Vec<Signature>,
-        commitment: Option<CommitmentConfig>,
+        config: Option<RpcSignatureStatusConfig>,
     ) -> RpcResponse<Vec<Option<TransactionStatus>>> {
         let mut statuses: Vec<Option<TransactionStatus>> = vec![];
```

Comment:
Let's put a reasonable limit on `signatures.len()`.   Otherwise the client can potentially run us out of memory.  256 signatures per batch seems like a ton for now